### ISSUE TITLE
torchvision shufflenetV2_x0.5 pytorch转ncnn后，模型输出差距巨大

### ISSUE THREAD

**Issue by freeman-l-20230331** [Tencent/ncnn#4681]
pytorch转onnx再转ncnn
pytorch和onnx的输出一致
ncnn输出差距巨大
同一张图片，pytorch和onnx输出：-1.9826340675354004, -1.5511791706085205, -5.220117568969727, -0.06643374264240265, 0.40736955404281616
ncnn输出：80.319229, -276.456573, -589.837280, -404.511475, -22.831861
转换用pnnx, convert model的网页, 自己代码都一样
pytorch前处理
transforms.ToTensor(),
transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
ncnn前处理
const float mean_vals[3] = {0.485f, 0.456f, 0.406f};
const float std_vals[3] = {1/0.229f, 1/0.224f, 1/0.225f};
const float norm_255[3] = {1/255.f, 1/255.f, 1/255.f};
const float empty[3] = {0.0f, 0.0f, 0.0f};
 in.substract_mean_normalize(empty, norm_255);
in.substract_mean_normalize(mean_vals, std_vals);
有没有大神帮忙看看，自己研究了一整天了没发现问题

**Comment by freeman-l-20230331** [comment]
我尝试了torchvision的pretrain model，也是一样的情况，ncnn输出和onnx差很远

**Comment by xellDart** [comment]
same result, any model for classification usin torchvision produces wrong results

**Comment by nihui** [comment]
转换和推理默认会自动使用 fp16，可以像这样禁用 fp16

```python
import torch
import torchvision.models as models

net = models.shufflenet_v2_x0_5(pretrained=True)
net.eval()

torch.manual_seed(0)
x = torch.rand(1, 3, 224, 224)

a = net(x)

# export torchscript
mod = torch.jit.trace(net, x)
mod.save("shufflenet_v2_x0_5.pt")

# torchscript to pnnx
import os
os.system("./pnnx shufflenet_v2_x0_5.pt inputshape=[1,3,224,224] fp16=0")

# pnnx inference
import shufflenet_v2_x0_5_pnnx
b = shufflenet_v2_x0_5_pnnx.test_inference()

# ncnn inference
import numpy as np
import ncnn
with ncnn.Net() as net:
    net.opt.use_fp16_storage = False
    net.load_param("shufflenet_v2_x0_5.ncnn.param")
    net.load_model("shufflenet_v2_x0_5.ncnn.bin")

    with net.create_extractor() as ex:
        ex.input("in0", ncnn.Mat(x.squeeze(0).numpy()).clone())

        _, out0 = ex.extract("out0")
        c = torch.from_numpy(np.array(out0)).unsqueeze(0)

# print torch/pnnx/ncnn result
print(a)
print(b)
print(c)
```

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
